### PR TITLE
Kamila/remove create profile

### PR DESCRIPTION
**Pull request by kamilababayeva**
This PR:

- simplify AuthenticationActivity and its tests
- remove CreateProfileActivity

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt [hunk 1]
```diff
@@ -21,13 +20,12 @@ import com.google.android.gms.common.api.ApiException
...
 import com.google.firebase.auth.AuthResult
 import com.google.firebase.auth.FirebaseAuth
-import com.google.firebase.auth.FirebaseUser
 import com.google.firebase.auth.GoogleAuthProvider
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
 import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
-
+import net.datafaker.Faker
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:28]: What is the purpose of DataFaker ?

Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:28]: Is it to create fake username ?

File: app/src/main/java/ch/sdp/vibester/database/DataGetter.kt [hunk 2]
```diff
@@ -52,7 +53,7 @@ class DataGetter @Inject constructor() {
     fun updateFieldInt(uid: String, fieldName: String, newVal: Int, method:String) {
         dbUserRef.child(uid).child(fieldName)
             .get().addOnSuccessListener { t ->
-                var finalVal  = checkValue(t, method, newVal)//newVal TO DELETE IF TESTS PASS, OTHERWISE ROLL BACK!
+                val finalVal  = checkValue(t, method, newVal)//newVal TO DELETE IF TESTS PASS, OTHERWISE ROLL BACK!
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/database/DataGetter.kt:56]: what is the purpose of this comment ? We should not keep it no ?

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt [hunk 3]
```diff
@@ -176,8 +171,7 @@ class AuthenticationActivityTest {
         onView(withId(R.id.password)).perform(ViewActions.typeText(password), closeSoftKeyboard())
         onView(withId(R.id.logIn)).perform(click())
 
-        Intents.intended(IntentMatchers.hasComponent(MyProfileActivity::class.java.name))
-        Intents.intended(IntentMatchers.hasExtra("email", username))
+        Intents.intended(IntentMatchers.hasComponent(ProfileActivity::class.java.name))
```
Comment (MaximeZmt) [inline app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:174]: Why not MyProfileActivity ?

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt [hunk 4]
```diff
@@ -191,13 +185,13 @@ class AuthenticationActivityTest {
...
+        every { mockAuthenticator.getCurrUserMail()} returns mockUser.email.toString()
+        every { mockAuthenticator.getCurrUID()} returns mockUser.uid
 
         onView(withId(R.id.username)).perform(ViewActions.typeText(username), closeSoftKeyboard())
         onView(withId(R.id.password)).perform(ViewActions.typeText(password), closeSoftKeyboard())
         onView(withId(R.id.createAcc)).perform(click())
-        Intents.intended(IntentMatchers.hasComponent(CreateProfileActivity::class.java.name))
-        Intents.intended(IntentMatchers.hasExtra("email", username))
-
+        Intents.intended(IntentMatchers.hasComponent(ProfileActivity::class.java.name))
```
Comment (MaximeZmt) [inline app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:194]: same here ? Why not MyProfileActivity

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt [hunk 5]
```diff
@@ -138,149 +147,106 @@ class AuthenticationActivity : AppCompatActivity() {
...
+
     /**
-     * A function that launches google sing in activity
+     * Start ProfileActivity
      */
-    private fun signInGoogle() {
-        val intent = googleSignInClient.signInIntent
-        startActivityForResult(intent, AUTHENTICATION_PERMISSION_CODE)
+    private fun startProfileActivity() {
+        val newIntent = Intent(this, ProfileActivity::class.java)
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:217]: Should go to MyProfileActivity

**Review by MaximeZmt — COMMENTED** [review]
Nice work !! 👍  Just left some comment below !

File: app/src/main/java/ch/sdp/vibester/database/DataGetter.kt [hunk 6]
```diff
@@ -100,12 +101,9 @@ class DataGetter @Inject constructor() {
      * @param callback function to be called when the the user has been created
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/database/DataGetter.kt:101]:
```suggestion
```

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt (hunk 1 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:28]: yes! I have a function for that uses it

File: app/src/main/java/ch/sdp/vibester/database/DataGetter.kt (hunk 2 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/database/DataGetter.kt:56]: It was not me who left it, i am not exaclty sure what it means... I will ask inva group

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt (hunk 3 above)
Comment (kamilababayeva) [reply app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:174]: my bad, thanks!

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt (hunk 4 above)
Comment (kamilababayeva) [reply app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:194]: you are right, will fix it

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt (hunk 5 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:217]: you are rightm will fix it

**Review by jiabaow — APPROVED** [review]
LGTM

**Review by MaximeZmt — APPROVED** [review]
Nice work 😀👍!

**Pull request merged**